src/libServer/Server.cpp

```diff
@@ -248,4 +304,392 @@ string Server::getHashrate()
+
+	return m_mediator.m_dsBlockChain.GetBlockCount().str();
+}
+
+string Server::getNumTransactions()
+{
+	LOG_MARKER();
+
+	
+	boost::multiprecision::uint256_t currBlock = m_mediator.m_txBlockChain.GetBlockCount() - 1;
+	LOG_MESSAGE("currBlock: "<<currBlock.str()<<" State: "<<m_BlockTxPair.first);
+	if(m_BlockTxPair.first < currBlock)
+	{
+		for(boost::multiprecision::uint256_t i = m_BlockTxPair.first + 1 ; i<=currBlock ; i++)
+		{
+			m_BlockTxPair.second += m_mediator.m_txBlockChain.GetBlock(i).GetHeader().GetNumTxs();
+
+		}
+	}
+	m_BlockTxPair.first = currBlock;
+
+	return m_BlockTxPair.second.str();
+
+}
+
+double Server::getTransactionRate()
+{
+	LOG_MARKER();
+
+	string numTxStr = Server::getNumTransactions();
+	boost::multiprecision::cpp_dec_float_50 numTxns(numTxStr);
+	LOG_MESSAGE("Num Txns: "<< numTxns);
+
+	numTxns = numTxns*1000000; // conversion from microseconds to seconds
+
+	//LOG_MESSAGE("TxBlockStart: "<<m_StartTimeTx<<" NumTxns: "<<numTxns);
+
+	if(m_StartTimeTx == 0) //case when m_StartTime has not been set 
+	{
+		try
+		{
+			TxBlock tx  = m_mediator.m_txBlockChain.GetBlock(1);
+			m_StartTimeTx = tx.GetHeader().GetTimestamp();
+		}
+		catch(const char* msg)
+		{
+			//cannot set
+			if(strcmp(msg,"Blocknumber Absent") == 0)
+			{
+				LOG_MESSAGE("No Tx Block has been mined yet");
+			}
+			return 0;
+		}
+		
+	}
+
+	boost::multiprecision::uint256_t TimeDiff = m_mediator.m_txBlockChain.GetLastBlock().GetHeader().GetTimestamp()-m_StartTimeTx;
+
+	
+
+	if(TimeDiff == 0)
+	{
+		return 0;
+	}
+
+	boost::multiprecision::cpp_dec_float_50 TimeDiffFloat = static_cast<cpp_dec_float_50>(TimeDiff);
+	boost::multiprecision::cpp_dec_float_50 ans = numTxns/TimeDiffFloat;
+
+	//LOG_MESSAGE("Rate: "<<ans);
+
+	return ans.convert_to<double>();
+	
+	
+}
+
+double Server::getDSBlockRate()
+{
+	LOG_MARKER();
+
+	string numDSblockStr = m_mediator.m_dsBlockChain.GetBlockCount().str();
+	boost::multiprecision::cpp_dec_float_50 numDs(numDSblockStr);
+
+	if(m_StartTimeDs == 0)
+	{
+		try
+		{
+			DSBlock dsb = m_mediator.m_dsBlockChain.GetBlock(1);
+			m_StartTimeDs = dsb.GetHeader().GetTimestamp();
+		}
+		catch(const char *msg)
+		{
+			if(strcmp(msg, "Blocknumber Absent") == 0)
+			{
+				LOG_MESSAGE("No DSBlock has been mined yet");
+			}
+			return 0;
+		}
+	}
+	boost::multiprecision::uint256_t TimeDiff = m_mediator.m_dsBlockChain.GetLastBlock().GetHeader().GetTimestamp() - m_StartTimeDs;
+
+	if(TimeDiff == 0)
+	{
+		LOG_MESSAGE("Wait till the second block");
+		return 0;
+	}
+	numDs = numDs*1000000;
+
+
+	boost::multiprecision::cpp_dec_float_50 TimeDiffFloat = static_cast<cpp_dec_float_50>(TimeDiff);
+	boost::multiprecision::cpp_dec_float_50 ans = numDs/TimeDiffFloat;
+
+	return ans.convert_to<double>();
+
+
+}
+
+double Server::getTxBlockRate()
+{
+	LOG_MARKER();
+
+	string numTxblockStr = m_mediator.m_txBlockChain.GetBlockCount().str();
+	boost::multiprecision::cpp_dec_float_50 numTx(numTxblockStr);
+	numTx = numTx*1000000;
```

Comment:
Why *1000000

Reply:
to convert microseconds into seconds

Reply:
the variable naming is confusing. One may think that it is number of txns on one glance. 
Change to a more meaningful name and put a comment for the conversion of ms to s.